### PR TITLE
Added multiple changes in project

### PR DESCRIPTION
**Pull request by Vitalinski**
- add  @imports
- delete unnecessary things, jsx endings in components
-move the components to the components file
- configure the prettier as a depository, add a configuration file and a script for formatting the project
- create a file with reusable constants and routes
- add css modules for each component 
- divide Slice.ts into parts
- change creating components on arrow function

### CODE REVIEW

File: src/components/Button/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.button {
+    font-weight: 700;
+    text-align: center;
+    color: #000;
+    cursor: pointer;
+
+
+    &:disabled{
+      border: 0.50px solid #a4a4a4;
```
Comment (odfrontendeveloper) [inline src/components/Button/index.module.scss]: you need to specify colors config in separate scss file and reuse as variables

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { FC } from 'react';
+import styles from '@/components/Button/index.module.scss';
+import { ButtonProps } from '@/store/interfaces';
+
+const Button: FC<ButtonProps> = (props) => {
+  return (
+    <button
+      onClick={props.onClick}
+      className={`${styles.button} ${styles[props.class]} ${styles[props.style]}`}
```
Comment (odfrontendeveloper) [inline src/components/Button/index.tsx]: use library "classnames"

File: src/components/Button/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      pointer-events: none;
+  background: #d7d7d7;
+  opacity: 0.5;
+    }
+  }
+
```
Comment (odfrontendeveloper) [inline src/components/Button/index.module.scss]: add scss format to prettier script

File: src/components/board/Card/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import Button from '@/components/Button';
+import { useStoreDispatch } from '@/store/store';
+import { openDelete, openModal, changeCurrentCard } from '@/store/cardsSlice';
+import styles from '@/components/board/Card/index.module.scss';
+import { CardProps } from '@/store/interfaces';
+
+const Card: FC<CardProps> = (props) => {
+  const dispatch = useStoreDispatch();
+
+  function deleteOpen() {
```
Comment (odfrontendeveloper) [inline src/components/board/Card/index.tsx]: use arrow functions for component included functions

File: src/components/modal/DeleteModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  function toDelete() {
+    dispatch(deleteCard());
+    cleaneAndClose();
+    dispatch(changeNotification('Card has been delited'));
+    setTimeout(() => {
+      dispatch(clearNotification());
+    }, 1000);
+  }
+  return (
+    <>
```
Comment (odfrontendeveloper) [inline src/components/modal/DeleteModal/index.tsx]: remove react fragment if it is not required

File: src/store/interfaces.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+interface Card {
```
Comment (odfrontendeveloper) [inline src/store/interfaces.ts]: I want to see file structure like this:

store
-cards
--index.ts
--cards.reducers.ts
--cards.types.ts
--cards.store.ts

etc.

File: src/components/modal/Input/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,29 @@
+.text{
```
Comment (odfrontendeveloper) [inline src/components/modal/Input/index.module.scss]: rename to Input.module.scss

"index" required for only root file in folder

File: src/components/modal/Input/index.module.scss [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    text-align: left;
+color: #616161;
+margin-bottom: 9px;
+  }
+.invalidText{
```
Comment (odfrontendeveloper) [inline src/components/modal/Input/index.module.scss]: use scss nesting (related to all scss syntax in project)

File: package.json [hunk 9]
```diff
@@ -7,7 +7,8 @@
     "dev": "vite",
     "build": "tsc && vite build",
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
```
Comment (Vitalinski) [inline package.json:10]: Check

File: src/components/Button/index.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    styles[props.class],
+    styles[props.style]
+  );
+  return (
+    <button
+      onClick={props.onClick}
+      className={buttonClass}
+      disabled={isWaiting || props.disabled}
+    >
+      {props.text}
```
Comment (odfrontendeveloper) [inline src/components/Button/index.tsx]: use props children instead of props text

File: src/App.tsx [hunk 11]
```diff
@@ -1,18 +1,17 @@
...
+      <Route path={ROUTES.START} element={<Login />} />
+      <Route path={ROUTES.LOGIN} element={<Login />} />
+      <Route path={ROUTES.DASHBOARD} element={<Dashboard />} />
+    </Routes>
+  );
+};
 
-export default App
+export default App;
+20;
```
Comment (odfrontendeveloper) [inline src/App.tsx]: what is this?

File: src/components/modal/Container/index.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import { FC } from 'react';
+import styles from '@/components/modal/Container/container.module.scss';
+import { ContainerProps } from '@/store/cards/cards.types';
+import { useSelector } from 'react-redux';
+import { RootState } from '@/store';
+
+const Container: FC<ContainerProps> = (props) => {
```
Comment (odfrontendeveloper) [inline src/components/modal/Container/index.tsx]: use props destructuring

File: src/components/modal/DeleteModal/index.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+  const [deleteCard] = useDeleteCardMutation();
+  const dispatch = useStoreDispatch();
+  const isDeleteOpen = useSelector((state: RootState) => state.cardAction.isDeleteOpen);
+  const isWaiting = useSelector((state: RootState) => state.cardAction.waitingMode);
+
+  const title = useSelector((state: RootState) => state.cardAction.currentCard.title);
+  const id = useSelector((state: RootState) => state.cardAction.currentCard.id);
+
+  useEffect(() => {
+    if (isDeleteOpen) {
```
Comment (odfrontendeveloper) [inline src/components/modal/DeleteModal/index.tsx]: you shouldn`t use changing styles like this, only scss

File: src/components/modal/DeleteModal/index.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'auto';
+    }
+  }, [isDeleteOpen]);
+
+  document.addEventListener('keydown', (event) => {
```
Comment (odfrontendeveloper) [inline src/components/modal/DeleteModal/index.tsx]: don`t use addEventListener in body of component, it cause infinity count of event listeners

File: src/components/modal/Modal/index.tsx [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+  const [description, setDescription] = useState(currentCard.description || '');
+  const [isTitleValid, setIsTitleValid] = useState(true);
+  const [isDescriptionValid, setIsDescriptionValid] = useState(true);
+  const isEdit = !!currentCard.title;
+  useEffect(() => {
+    setTitle(currentCard.title !== undefined ? currentCard.title : '');
+    setDescription(currentCard.description !== undefined ? currentCard.description : '');
+  }, [currentCard]);
+  useEffect(() => {
+    if (isModalOpen) {
```
Comment (odfrontendeveloper) [inline src/components/modal/Modal/index.tsx]: code duplicate?

File: src/store/cards/card.reducers.ts [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+  state.currentCard.description = info.payload.description;
+};
+const InitialiseUser = (state: RootState, email: { payload: string }) => {
+  localStorage.setItem('userEmail', JSON.stringify(email.payload));
+  state.userEmail = email.payload;
+};
+const ChangeNotification = (state: RootState, info: { payload: string }) => {
+  state.notification = info.payload;
+};
+const ClearNotification = (state: RootState) => {
```
Comment (odfrontendeveloper) [inline src/store/cards/card.reducers.ts]: don`t use capitalize for functions, if they are not a components

File: src/store/cards/cards.apiCalls.ts [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+import { Card, CurrentCard } from '@/store/cards/cards.types';
+
+export const cardsAPI = createApi({
+  reducerPath: 'cardsAPI',
+  baseQuery: fetchBaseQuery({
+    baseUrl: 'http://167.172.185.32:8089/cards/',
+  }),
+  tagTypes: ['Card'],
+  endpoints: (builder) => ({
+    getCards: builder.query<Card[], string>({
```
Comment (odfrontendeveloper) [inline src/store/cards/cards.apiCalls.ts]: move endpoints to a separate file

File: src/components/Header/index.tsx [hunk 18]
```diff
@@ -0,0 +1,27 @@
+import Button from '@/components/Button';
+import { Link } from 'react-router-dom';
+import { useStoreDispatch } from '@/store';
+import { toClearState } from '@/store/cards';
+import styles from '@/components/Header/header.module.scss';
+import { ROUTES } from '@/store/routes';
+import { FC } from 'react';
+const Header: FC = () => {
+  const userEmail: string = localStorage.getItem('userEmail') || '';
```
Comment (odfrontendeveloper) [inline src/components/Header/index.tsx]: move working with local storage to separate service

File: src/components/board/Card/index.tsx [hunk 19]
```diff
@@ -0,0 +1,39 @@
+import { FC } from 'react';
+import Button from '@/components/Button';
+import { useStoreDispatch } from '@/store';
+import { openDelete, openModal, changeCurrentCard } from '@/store/cards';
+import styles from '@/components/board/Card/card.module.scss';
+import { CardProps } from '@/store/cards/cards.types';
+
+const Card: FC<CardProps> = (props) => {
```
Comment (odfrontendeveloper) [inline src/components/board/Card/index.tsx]: wrap to React.memo

File: src/components/modal/DeleteModal/index.tsx [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+    if (isWaiting) return;
+    dispatch(closeModal());
+  };
+  const toDelete = async () => {
+    if (typeof id === 'number') {
+      dispatch(changeWaitingMode(true));
+      try {
+        await deleteCard(id).unwrap();
+        dispatch(changeWaitingMode(false));
+        dispatch(changeNotification('Card has been deleted'));
```
Comment (odfrontendeveloper) [inline src/components/modal/DeleteModal/index.tsx]: move text messages to constants

File: src/components/modal/DeleteModal/index.tsx [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+  closeModal,
+} from '@/store/cards';
+import Container from '@/components/modal/Container';
+import styles from '@/components/modal/DeleteModal/delete.module.scss';
+import { useDeleteCardMutation } from '@/store/cards/cards.apiCalls';
+import Overlay from '@/components/modal/Overlay';
+const DeleteModal: FC = () => {
+  const [deleteCard] = useDeleteCardMutation();
+  const dispatch = useStoreDispatch();
+  const isDeleteOpen = useSelector((state: RootState) => state.cardAction.isDeleteOpen);
```
Comment (odfrontendeveloper) [inline src/components/modal/DeleteModal/index.tsx]: selectors should be defined in separate file

File: src/components/modal/DeleteModal/index.tsx [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+    if (event.key === 'Escape' && isDeleteOpen) {
+      cleaneAndClose();
+    }
+  });
+  const cleaneAndClose = () => {
+    if (isWaiting) return;
+    dispatch(closeModal());
+  };
+  const toDelete = async () => {
+    if (typeof id === 'number') {
```
Comment (odfrontendeveloper) [inline src/components/modal/DeleteModal/index.tsx]: !!id

File: src/components/modal/Modal/index.tsx [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'auto';
+    }
+  }, [isModalOpen]);  
+  const addNewCard = async () => {
+    const newCard = {
```
Comment (odfrontendeveloper) [inline src/components/modal/Modal/index.tsx]: use RTK query provided variables for loading

File: src/store/cards/cards.types.ts [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+  closeBtn: boolean;
+  onClick?: () => void;
+  children?: React.ReactNode;
+}
+
+interface ButtonProps {
+  disabled?: boolean;
+  text: string;
+  style: string;
+  class: string;
```
Comment (odfrontendeveloper) [inline src/store/cards/cards.types.ts]: use fixed names for button types (yellow, white, etc.)

File: src/store/cards/index.ts [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+  ChangeNotification,
+  ClearNotification,
+  CloseModal,
+  OpenDelete,
+  OpenModal,
+} from './card.reducers';
+
+const currentCard: CurrentCard = {};
+
+const initialState: RootState = {
```
Comment (odfrontendeveloper) [inline src/store/cards/index.ts]: initial state, reducers to a separate file

**Pull request merged**